src/main/java/io/vertx/core/http/impl/Http2ServerConnection.java

```diff
@@ -86,44 +87,35 @@ public HttpServerMetrics metrics() {
     return metrics;
   }
 
-  private static boolean isMalformedRequest(Http2Headers headers) {
+  private static HostAndPort isMalformedRequest(Http2Headers headers) {
     if (headers.method() == null) {
-      return true;
+      return INVALID_REQUEST_SENTINEL;
     }
     String method = headers.method().toString();
```

Comment:
`*.method()/*.authority() ` etc etc looks like plain field access but they are not: are always lookup into a map, which is not cheap and we can just save in local fields them

Reply:
can't we have an header implementation that does that ?